collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -446,17 +446,13 @@ private void loadForm() {
             saveToDiskTask = (SaveToDiskTask) data;
         } else if (data == null) {
             if (!newForm) {
-                if (getFormController() != null) {
+                if (getFormController(true) != null) {
                     refreshCurrentView();
                 } else {
                     Timber.w("Reloading form and restoring state.");
-                    // we need to launch the form loader to load the form
-                    // controller...
-                    formLoaderTask = new FormLoaderTask(instancePath,
-                            startingXPath, waitingXPath);
-                    // TODO: this doesn' work (dialog does not get removed):
-                    // showDialog(PROGRESS_DIALOG);
```

Comment:
I encounter this "TODO" issue when hard-coding this execution path (dialog does not get removed, it just remains spinning until I press the cancel button).

Reply:
That's not true (the comment) I tested it and the dialog is removed properly. It's a very old code so probably it was something wrong back then.

Reply:
I fixed that problem introducing DialogFragment instead.